include/materials/DamagePlasticityStressUpdate.h

```diff
@@ -0,0 +1,184 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/*                       BlackBear                              */
+/*                                                              */
+/*           (c) 2017 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#pragma once
+
+#include "MultiParameterPlasticityStressUpdate.h"
+
+class DamagePlasticityStressUpdate : public MultiParameterPlasticityStressUpdate
+{
+public:
+  static InputParameters validParams();
+  DamagePlasticityStressUpdate(const InputParameters & parameters);
+
+  /**
+   * Does the model require the elasticity tensor to be isotropic?
+   */
+  bool requiresIsotropicTensor() override { return true; }
+
+protected:
+  virtual void initQpStatefulProperties() override;
+  virtual void finalizeReturnProcess(const RankTwoTensor & rotation_increment) override;
+
+private:
+  const Real _f_tol;
```

Comment:
I hate to say this, but we should have doxygen comments describing what all the member variables and non-overwritten methods are.